pyiceberg/catalog/dynamodb.py

```diff
@@ -98,6 +99,7 @@ def __init__(self, name: str, **properties: str):
         session = boto3.Session(
             profile_name=properties.get(DYNAMODB_PROFILE_NAME),
             region_name=get_first_property_value(properties, DYNAMODB_REGION, AWS_REGION),
+            botocore_session=properties.get(DEPRECATED_BOTOCORE_SESSION),
```

Comment:
Do we want to emit a deprecation message? I feel like there is no strong consensus on how to move this forward, and I would also love to get @HonahX's input on it. Same for Glue below

Reply:
im ok with leaving it as is now, and deferring to the next release